test/suites/drift/suite_test.go

```diff
@@ -337,6 +337,115 @@ var _ = Describe("Drift", Label("AWS"), func() {
 		Entry("DetailedMonitoring Drift", "DetailedMonitoring", v1alpha1.AWSNodeTemplateSpec{DetailedMonitoring: awssdk.Bool(true)}),
 		Entry("AMIFamily Drift", "AMIFamily", v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{AMIFamily: awssdk.String(v1alpha1.AMIFamilyBottlerocket)}}),
 	)
+	Context("Drift Failure", func() {
+		It("should not continue to drift if a node never registers", func() {
+			// launch a new machine
+			var numPods int32 = 2
+			dep := test.Deployment(test.DeploymentOptions{
+				Replicas: 2,
+				PodOptions: test.PodOptions{
+					ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "inflate"}},
+					PodAntiRequirements: []v1.PodAffinityTerm{{
+						TopologyKey: v1.LabelHostname,
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{"app": "inflate"},
+						}},
+					},
+				},
+			})
+			env.ExpectCreated(dep, nodeTemplate, provisioner)
+			env.EventuallyExpectNodeCount("==", 2)
+
+			// Drift the machine with bad configuration
+			parameter, err := env.SSMAPI.GetParameter(&ssm.GetParameterInput{
+				Name: awssdk.String("/aws/service/ami-amazon-linux-latest/amzn-ami-hvm-x86_64-ebs"),
+			})
+			Expect(err).ToNot(HaveOccurred())
+			nodeTemplate.Spec.AMISelector = map[string]string{"aws::ids": *parameter.Parameter.Value}
+			env.ExpectCreatedOrUpdated(nodeTemplate)
+
+			// Should see the machine has drifted
+			machines := env.EventuallyExpectCreatedMachineCount("==", int(numPods))
+			Eventually(func(g Gomega) {
+				for _, machine := range machines {
+					g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(machine), machine)).To(Succeed())
+					g.Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted).IsTrue()).To(BeTrue())
+				}
+			}).Should(Succeed())
+
+			// Expect machine To get cordoned
+			cordonedNodes := []*v1.Node{}
+			nodes := env.EventuallyExpectNodeCount("==", int(numPods))
+			Eventually(func(g Gomega) {
+				cordonedNodes = lo.Filter(nodes, func(node *v1.Node, _ int) bool {
+					g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(node), node))
+					return node.Spec.Unschedulable == true
+				})
+				g.Expect(len(cordonedNodes)).To(BeNumerically("==", 1))
+			}).Should(Succeed())
+
+			// Drift should fail and original node should be uncordoned
+			// TODO: reduce timeouts when deprovisioning waits are factored out
+			machineList := &v1alpha5.MachineList{}
+			Eventually(func(g Gomega) {
+				g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(cordonedNodes[0]), cordonedNodes[0]))
```

Comment:
This is another case where you need a local copy. If this call ever fails, you risk mutating the data that you are trying to use as your static set of data with the dynamic set of data that you are testing against.